webapp/components/generic/SearchableInput/SearchableInput.spec.js

```diff
@@ -32,81 +32,79 @@ describe('SearchableInput.vue', () => {
   }
 
   describe('mount', () => {
-    describe('testing custom functions', () => {
-      let select
+    let select
 
-      beforeEach(() => {
-        select = wrapper.find('.ds-select')
-        select.trigger('focus')
-        select.element.value = 'abcd'
-      })
+    beforeEach(() => {
+      select = wrapper.find('.ds-select')
+      select.trigger('focus')
+      select.element.value = 'abcd'
+    })
 
-      it('opens the select dropdown when focused on', () => {
-        expect(wrapper.find('.is-open').exists()).toBe(true)
-      })
+    it('opens the dropdown when focused', () => {
+      expect(wrapper.find('.ds-select-dropdown').isVisible()).toBe(true)
+    })
 
-      it('opens the select dropdown and blurs after focused on', () => {
-        select.trigger('blur')
-        expect(wrapper.find('.is-open').exists()).toBe(false)
-      })
+    it('closes the dropdown when blurred', () => {
+      select.trigger('blur')
+      expect(wrapper.find('.ds-select-is-open').exists()).toBe(false)
+    })
 
-      it('is clearable', () => {
-        select.trigger('input')
-        select.trigger('keyup.esc')
-        expect(wrapper.find('.is-open').exists()).toBe(false)
-      })
+    it('closes the dropdown when cleared with esc key', () => {
+      select.trigger('input')
+      select.trigger('keyup.esc')
+      expect(wrapper.find('.ds-select-is-open').exists()).toBe(false)
+    })
 
-      it('changes the unprocessedSearchInput as the value changes', () => {
-        select.trigger('input')
-        expect(select.element.value).toBe('abcd')
-      })
+    it('changes the unprocessedSearchInput as the value changes', () => {
+      select.trigger('input')
+      expect(select.element.value).toBe('abcd')
+    })
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-01-15T16:25:47Z" title="Wednesday, January 15th 2020, 5:25:47 pm +01:00">Jan 15, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -32,81 +32,79 @@ describe('SearchableInput.vue', () => {
   }
 
   describe('mount', () => {
-    describe('testing custom functions', () => {
-      let select
+    let select
 
-      beforeEach(() => {
-        select = wrapper.find('.ds-select')
-        select.trigger('focus')
-        select.element.value = 'abcd'
-      })
+    beforeEach(() => {
+      select = wrapper.find('.ds-select')
+      select.trigger('focus')
+      select.element.value = 'abcd'
+    })
 
-      it('opens the select dropdown when focused on', () => {
-        expect(wrapper.find('.is-open').exists()).toBe(true)
-      })
+    it('opens the dropdown when focused', () => {
+      expect(wrapper.find('.ds-select-dropdown').isVisible()).toBe(true)
+    })
 
-      it('opens the select dropdown and blurs after focused on', () => {
-        select.trigger('blur')
-        expect(wrapper.find('.is-open').exists()).toBe(false)
-      })
+    it('closes the dropdown when blurred', () => {
+      select.trigger('blur')
+      expect(wrapper.find('.ds-select-is-open').exists()).toBe(false)
+    })
 
-      it('is clearable', () => {
-        select.trigger('input')
-        select.trigger('keyup.esc')
-        expect(wrapper.find('.is-open').exists()).toBe(false)
-      })
+    it('closes the dropdown when cleared with esc key', () => {
+      select.trigger('input')
+      select.trigger('keyup.esc')
+      expect(wrapper.find('.ds-select-is-open').exists()).toBe(false)
+    })
 
-      it('changes the unprocessedSearchInput as the value changes', () => {
-        select.trigger('input')
-        expect(select.element.value).toBe('abcd')
-      })
+    it('changes the unprocessedSearchInput as the value changes', () => {
+      select.trigger('input')
+      expect(select.element.value).toBe('abcd')
+    })
```


I know you didn't add this, but this is not testing anything in my mind, and can be simply removed...originally this was testing `wrapper.vm.unprocessedSearchInput`, but @roschaefer said to avoid this cause it can cause false negatives.
maybe we don't need this `unprocessedSearchInput` at all... but we can certainly keep refactoring in the future, especially when we migrate the `Select` component...

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2020-01-22T07:46:04Z" title="Wednesday, January 22nd 2020, 8:46:04 am +01:00">Jan 22, 2020</time>_

---


Yeah, I was wondering about this and didn't really get what it was supposed to test. Let's keep refactoring! :grin: